src/core/Menu/MenuItem.tsx

```diff
@@ -119,12 +171,42 @@ export const MenuItem = React.forwardRef<HTMLLIElement, MenuItemProps>(
           <div className='iui-menu-label'>{children}</div>
           {sublabel && <div className='iui-menu-description'>{sublabel}</div>}
         </span>
+        {!badge && subMenuItems.length > 0 && (
+          <SvgCaretRightSmall className='iui-icon' />
+        )}
         {badge &&
           React.cloneElement(badge, {
             className: cx(badge.props.className, 'iui-icon'),
           })}
       </li>
     );
+
+    return subMenuItems.length === 0 ? (
+      listItem
+    ) : (
+      <MenuItemContext.Provider value={{ ref: menuItemRef }}>
+        <Popover
+          placement='right-start'
+          visible={isSubmenuVisible}
+          content={
+            <div
+              onMouseLeave={() => setIsSubmenuVisible(false)}
+              onKeyDown={(e) => {
```

Comment:
What for this was added? Now this one is being called together with the left key down handler above (where we focus and that should close menu anyway...).
Also, while playing around, still managed to get this one somehow
![image](https://user-images.githubusercontent.com/81580355/126460718-978da8f9-8207-4caa-9bee-070ad5e591a5.png)

Reply:
I have fixed double opened sub menus with the onBlur event addition on a new div..
Now the only issue left is that the Menu component losses the correct `focusedIndex` when you start playing around with mouse + keyboard combo...
But honestly, I am a little lost, how that should even work...

Reply:
I was trying to look for any examples how keyboard nav should work here, but.. Not much I could find.
There was this:
https://codepen.io/vj/pen/rOgPEY
Which works using tabbing (might be easier, but I am not sure if it is better...).
Also w3 has some article about it, but not much helpful in our case though.
https://www.w3.org/WAI/tutorials/menus/flyout/
And noticed SmashingMagazine article and this attribute `aria-haspopup` which we do not use for menu for some reason...
https://www.smashingmagazine.com/2017/11/building-accessible-menu-systems/

Reply:
Removed onKeyDown as it's not needed now that you're closing submenu in onBlur.

> Now the only issue left is that the Menu component losses the correct `focusedIndex` when you start playing around with mouse + keyboard combo... But honestly, I am a little lost, how that should even work...

That's fine, I don't think we need to make it perfect for v1. Mouse + keyboard is a weird use case anyway.

> attribute `aria-haspopup` which we do not use for menu for some reason...

Added.